docs-ref-conceptual/format-output-azure-cli.md

```diff
@@ -181,44 +186,117 @@ The next example shows how `tsv` output can be piped to other commands in bash.
+
+Reset a service principal credential returning output in the default json format:
+
+```azurecli-interactive
+# reset service principal credentials returning results to the console and the log
+az ad sp credential reset --id myServicePrincipalID --output json
+```
+
+Console output showing the new password in the console, and also visible in the log:
+
+```output
+{
+  "appId": "myServicePrincipalID",
+  "password": "myServicePrincipalNewPassword",
+  "tenant": "myTenantID"
+}
+```
+
+A better solution is to return security information to a variable. This example _does not_ write the service principal password to the log. When testing, use the `echo` command to test the output of your variable, but understand that `echo` writes to the log.
+
+```azurecli-interactive
+# reset service principal credentials returning results to a variable
+myNewPassword=$(az ad sp credential reset --id myServicePrincipalID --query password --output tsv)
+```
+
+For more examples on storing output to a variable, see [Use the Azure CLI successfully - pass values to another command](./use-cli-effectively.md#pass-values-to-another-command). To learn more about the `--query` parameter, see [How to query Azure CLI command output](./query-azure-cli.md).
+
 ## Set the default output format
 
-Use the `az config set` command to set up your environment and establish default settings for output formats. The default output format is `json`.
+Azure CLI commands provide output that can be controlled in two ways:
+
+|Output control | How-to |
+|-|-|
+|Global setting| Specify a default output format using [az config set](./azure-cli-configuration.md#cli-configuration-values-and-environment-variables).
+|Command parameter| Override the default setting using a reference command's `--output` parameter.
+
+The default output for the Azure CLI is `json`. Set the default output to `none` when console output and logging is not needed.
 
 ```azurecli
-az config set core.output=<format>
+az config set core.output=none
 ```
 
-```output
-Welcome to the Azure CLI! This command will guide you through logging in and setting some default values.
+You can overwrite the default output of any Azure CLI reference command by using the `--output` parameter. Here is a script of commands that alter and test command output:
 
-Your settings can be found at /home/defaultuser/.azure/config
-Your current configuration is as follows:
+```azurecli
+# set your default output to table
+az config set core.output=table
 
-  ...
+# show your active subscription in table format
+# notice how only a subset of properties are returned in the table
+az account show
 
-Do you wish to change your settings? (y/N): y
+# override your table default and show your active subscription in jsonc format
+az account show --output jsonc
 
-What default output format would you like?
```

Comment:
Note to reviewers: This section was removed because this command no longer prompts for a numeric choice.